### ISSUE TITLE
Invalid parameter: redirect_uri despite * Valid Redirect URIs being set correctly

### ISSUE THREAD

**Issue by haddly** [keycloak/keycloak#15304]
### Area

admin/cli

### Describe the bug

Keycloak has been running fine until I believe a new update happened in the last few weeks. Now, when logging of, a screen appears with: 
We are sorry...
Invalid parameter: redirect_uri
In my admin console, the correct uris have been set with production uri and localhost for dev:
http://localhost:3001/*
Is there any new setting i need to make sure this works? 

### Version

19.0.3

### Expected behavior

Run logout --> User is logged out and redirects to LandingPage. On development this is localhost:3001.

### Actual behavior

Click logout --> Keycloak screen appears with 'we are sorry' message
Only way to logout now is to remove the following part of the uri in the browser and hit enter:
?redirect_uri=http%3A%2F%2Flocalhost%3A3001%2F

### How to Reproduce?

run keycloak logout.

**Comment by lexcao** [comment]
Hi, have you checked the upgrading guide?
https://www.keycloak.org/docs/latest/upgrading/index.html#openid-connect-logout

The logout now needs to configure the `post_logout_redirect_uri` from Admin for the client.
Or you could just leave `+` for extending the current `redirect_uris`

**Comment by kolonuk** [comment]
+1 here, fresh install.

My keycloak error log is stating that post_logout_redirect_uri should be used instead.

Unsure if this is something I should be able to change, as there isn't an option on the client screen.

**Comment by lexcao** [comment]
Hi @kolonuk 
Maybe you should configure the `Valid post logout redirect URIs` like below.
* you could just leave + for extending the current redirect_uris

<img width="720" alt="image" src="https://user-images.githubusercontent.com/14901139/200888267-913192d5-6e15-40bc-8de0-879b10cfeaaa.png">

**Comment by kolonuk** [comment]
OK, I think this issue falls into one of two categories:

1. Developer of a product that uses keycloak for authentication, and therefore can make changes to the source code of their own app, but is otherwise ignorant of or not following keycloak changes becuase they don't use it every day or have no testing facility before release.
2. End user/admin, which has installed some 3rd party software and keycloak, but is experiencing errors such as this in the keycloak logs.

In light of realising this understanding, and finding out that I sit into option 2 above:
1. I will go back to the software developer and recommend they change the URI they are calling for logout.
2. Suggest that you make errors in keycloak have a better indication as to whether it is an end user issue (ie incorrect logins, incorrect setup etc) or it is an integration issue with the 3rd party software using keycloak for authentication.

TLDR; It's not your or my issue, it's Documize's problem!

**Comment by Cracky5457** [comment]
I can confirm this bug.

I don't think keycloak should introduce this kind of broker change, our url have a redirect_uri param for the logout to redirect to the homepage of an application, but filling`Valid redirect URIs ` with this param doesn't solve the issue, and with + it kinda solve it only if we remove redirect_uri from the logout url and then it stay on KC logout page and doesn't redirect to the webhome of the user application.

Strange behaviour and regression here

**Comment by mabartos** [comment]
@haddly Thanks for the report and sorry for the late response. 

This issue is getting stale, and from my understanding, it doesn't reflect any bug in Keycloak(in the more recent versions). Furthermore, the potential workaround is described here. I don't see any improvement we can make here. 

Feel free to reopen the issue if you encounter a similar problem in the most recent version of Keycloak.

**Comment by idc77** [comment]
In my case, I have a flutter app, which uses AppAuth, which used to work, and now doesn't anymore.
Everything relevant is set to *

**Comment by worldomkar** [comment]
Adding a side-note here for those stumbling upon this stale issue:

I have spent some hours in narrowing this down, which seemed a keycloak issue.

In a boilerplate express + node + keycloak-connect + express-session, after correctly redirecting to google login, selecting google user, and coming back to keycloak auth url, this error occurs.

The reason being due to some unknown (unlikely config issue) - probably a bug in the keycloak-connect library, it generates different URI everytime, sometimes, with multiple ...&iss=...&iss=... repeated values.

For me fix was to append a "*" at the end of the redirect URL: http://localhost:8080/*

This works for DEVELOPMENT, but nobody can't be happy for this in prod. I would need to dig more "compatible" production related lib (or its config).
